### DISCUSSION TITLE
Add a Combo status HUD Element

### DISCUSSION THREAD

**Discussion by moonleay** [ppy/osu#30380] [Ideas]
I think it would be a good idea to add a "Combo Status" element, which displays, if you are currently doing a Perfect run, FC'ing, or how may times you missed.
I know there is already the "Judgement Counter" element but it does not allow for a quick check during runs.

I made a quick PoC:

https://github.com/user-attachments/assets/cc0e22a5-5cd0-4c5b-b960-ce9e0c95743c

**Comment by Joehuu** [comment]
Doesn't the rank display already show this? SS indicates a perfect, S indicates an FC and >95%, and A indicates a miss or <95%. Combine it with the judgment counter, and you get the miss counter + other judgements.

POC also seems to change from PRFCT to FC after the first slider is hit perfectly?

**Reply by moonleay** [reply]
> Doesn't the rank display already show this?

It kinda does and it kinda doesn't. You are right, it can display that kind of info in combination with the judgment counter, but that requires looking at two places and is a bigger mental load then just looking at a HUD element and getting an answer, which is annoying when you are trying to concentrate on a map. It also does not work, if you are <95%.


> POC also seems to change from PRFCT to FC after the first slider is hit perfectly?

Oh, you are right. I think I overlooked that.
EDIT: I fixed it.

https://github.com/user-attachments/assets/4e0835cc-6f27-40a8-9858-f711db23dd35

**Reply by Joehuu** [reply]
I dunno. There are many other visual/auditory cues already and you don't have to see the HUD at all:
- if you see an ok or meh judgment, it's not perfect (you can clearly see this with the argon pro skin or skinning out the great judgement)
- if you miss or hear a combo break, it's not an FC

Might make sense for a viewer if you're streaming, but I believe the judgement counter is plenty enough based on my reasoning above.

I also think we try to avoid terms like FC because new users may not know that term and it would need to be "Full Combo", which is too long. The term/acroym is not used anywhere in the website or game.

https://github.com/ppy/osu/issues/8413 might improve your case, though.
